x11-misc/slim/slim-1.3.9.ebuild

```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake pam systemd readme.gentoo-r1
+
+DESCRIPTION="Simple Login Manager resurrected"
+HOMEPAGE="https://slim-fork.sourceforge.io/"
+SRC_URI="mirror://sourceforge/project/${PN}-fork/${P}.tar.gz"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~sparc ~x86"
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="branding pam"
+
+RDEPEND="media-libs/libpng:0=
+	media-libs/libjpeg-turbo:=
+	x11-libs/libXmu
+	x11-libs/libX11
+	x11-libs/libXpm
+	pam? ( x11-libs/libXext )
```

Comment:
The rest of the dependencies need to be sorted too, for example `x11-libs/libXft` comes before `x11-libs/libXpm`, `x11-apps/` before `x11-libs/` and so on.

Reply:
OK, though it raises a separate question, which is a matter of preference, I think:
The pam USE flag currently controls whether slimlock is built. This follows the logic in the upstream CMakeLists from when Nobuhiro-san added it (at V1.3.6). That was, I think, an error - the slimlock code needs PAM and so should not be built _without_ USE_PAM, but it's not required to build it because of PAM. This logic has changed upstream from V1.4.0 but I've retained the old behaviour through the ebuild. It might be considered better to introduce a different (package-specific) USE flag to control it, then encode that this USE flag needs USE=pam first. Of course, that would also be a change of behaviour, so might need a message, too... and I'm not sure it's really necessary.

Reply:
Do you see any value in building with `pam` but without slimlock?  Does slimlock require any extra dependencies, or does it take a long time to build?  If not, no need to add an extra USE flag, tying slimlock to the same knob as pam sounds reasonable.

Reply:
I don't think there's any real benefit other than "removal of cruft" (since I suspect very few people actually use slimlock). It adds only one dependency (libXext) but from the looks of things that's actually already an indirect dependency (via libXft?) and the build step for slimlock is pretty quick (one C++ file and a link).
So yes, the current approach feels pragmatically sensible.